Assets/MixedRealityToolkit/UX/Scripts/BoundingBoxes/BoundingBoxRig.cs

```diff
@@ -0,0 +1,470 @@
+            }
+
+            set
+            {
+                scaleHandleMaterial = value;
+            }
+        }
+
+        public Material RotateHandleMaterial
+        {
+            get
+            {
+                return rotateHandleMaterial;
+            }
+
+            set
+            {
+                rotateHandleMaterial = value;
+            }
+        }
+
+        public Material InteractingMaterial
+        {
+            get
+            {
+                return interactingMaterial;
+            }
+
+            set
+            {
+                interactingMaterial = value;
+            }
+        }
+
+        public void Activate()
+        {
+            ShowRig = true;
+        }
+
+        public void Deactivate()
+        {
+            ShowRig = false;
+        }
+
+        public void FocusOnHandle(GameObject handle)
+        {
+            if (handle != null)
+            {
+                for (int i = 0; i < rotateHandles.Length; ++i)
+                {
+                    rotateHandles[i].SetActive(rotateHandles[i].gameObject == handle);
+                }
+                for (int i = 0; i < cornerHandles.Length; ++i)
+                {
+                    cornerHandles[i].SetActive(cornerHandles[i].gameObject == handle);
+                }
+            }
+            else
+            {
+                for (int i = 0; i < rotateHandles.Length; ++i)
+                {
+                    rotateHandles[i].SetActive(true);
+                }
+                for (int i = 0; i < cornerHandles.Length; ++i)
+                {
+                    cornerHandles[i].SetActive(true);
+                }
+            }
+        }
+
+        private void Start()
+        {
+            objectToBound = this.gameObject;
+
+            boxInstance = Instantiate(BoundingBoxPrefab) as BoundingBox;
+            boxInstance.Target = objectToBound;
+            boxInstance.FlattenPreference = flattenedAxis;
+
+            BuildRig();
+
+            appBarInstance = Instantiate(appBarPrefab) as AppBar;
+            appBarInstance.BoundingBox = boxInstance;
+
+            boxInstance.IsVisible = false;
+        }
+        private void Update()
+        {
+            if (destroying == false && ShowRig)
+            {
+                UpdateBoundsPoints();
+                UpdateHandles();
+            }
+        }
+
+        private void OnDestroy()
+        {
+            destroying = true;
+            ShowRig = false;
+            ClearHandles();
+        }
+
+        private void UpdateBoundsPoints()
+        {
+            handleCentroids = GetBounds();
+        }
+
+        private void CreateHandles()
+        {
+            ClearHandles();
+            UpdateCornerHandles();
+            UpdateRotateHandles();
+            ParentHandles();
+            UpdateHandles();
+        }
+
+        private void UpdateCornerHandles()
+        {
+            handleCentroids = handleCentroids ?? GetBounds();
```

Comment:
not sure ?? works the way you expect for `UnityEngine.Object` types.
Specifically checking `!=null` is best practice here.

Reply:
Due to the current state of Unity, we do not advise using the null coalescing operator.  Please revert to the old if == null check.

Reply:
will do